### PR TITLE
Graceful shutdown timeout

### PR DESCRIPTION
**Pull request by elliotfehr**
**Description of the change:**
Adds a `--drain-time` to ansible based operators

**Motivation for the change:**
This is configurable in the manager but not possible to override using environment variables or CLI flags for roles that might take more than the 30 second default to complete.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: changelog/fragments/ansible-drain-time.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+entries:
+  - description: >
+      For Ansible-based operators, allow passing a --drain-time to configure the duration the manager should wait before stopping.
```
Comment (estroz) [inline changelog/fragments/ansible-drain-time.yaml:3]: Why call this --drain-time? Does this terminology come from somewhere?

**Review by estroz — CHANGES_REQUESTED** [review]
In general I'm ok with allowing flag overrides of operator values. I will say, now that controller-runtime supports a [config](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/config), flags should be added in controller-runtime using component-base's [`cli/flag`](https://github.com/kubernetes/component-base/blob/master/cli/flag) lib to get these flags for free. I'm ok adding a one-off flag like this now as long as the name aligns with what would be added in controller-runtime so config flags can be switched in later without a breaking change.

File: changelog/fragments/ansible-drain-time.yaml (hunk 1 above)
Comment (elliotfehr) [reply changelog/fragments/ansible-drain-time.yaml:3]: No real reason or preference here. Any suggestions for what to name it?

Comment (estroz) [reply changelog/fragments/ansible-drain-time.yaml:3]: `--graceful-shutdown-timeout`

Comment (elliotfehr) [reply changelog/fragments/ansible-drain-time.yaml]: Updated all references!

**Review by estroz — APPROVED** [review]
/lgtm

**Pull request merged**